src/java_bytecode/java_bytecode_instrument.cpp

```diff
@@ -354,7 +354,17 @@ void java_bytecode_instrumentt::prepend_instrumentation(
 }
 
 /// Augments `expr` with instrumentation in the form of either
-/// assertions or runtime exceptions
+/// assertions or runtime exceptions.
+/// The list of runtime exceptions that can be thrown without having been
+/// declared explicitly using 'throw new' is:
+/// ArithmeticException
+/// ArrayIndexOutOfBoundsException
+/// ClassCastException
+/// NegativeArraySizeException
+/// NullPointerException
+/// If this list grows in the future, any new classes will have to both be
+/// handled by this function and added to the list of classes that should
+/// always be loaded (by the Java class loader and by lazy methods).
```

Comment:
Looks good - but could you put explicit source locations (e.g. `class_name::function`) so people know where to look